dcap/types/src/error.rs

```diff
@@ -16,7 +16,7 @@ use serde::{Deserialize, Serialize};
 #[cfg_attr(feature = "serde", derive(Deserialize, Serialize))]
 #[non_exhaustive]
 #[repr(u32)]
-pub enum Error {
+pub enum SgxError {
```

Comment:
Another option is to rename the module `sgx_error.rs` and keep the enum named `Error` then rename in the public API like the `Error` -> `Quote3Error` that was happening.

This naming issue was noticed as the Quote3 type started to gain fallible behavior. It feels off that the Quote3Error doesn't correlate to a failure of the Quote3 type. The Quote3Error naming was mimicing the SGX SDK's `quote3_error_t`.

Reply:
This has been further renamed to `QlError` leaving full comment open for context/discussion.